ast/src/parsed/mod.rs

```diff
@@ -929,6 +929,7 @@ impl<R> Expression<R> {
     }
 
     /// Returns true if the expression contains a reference to a next value
+    // TODO: Is it fine that this does not check references to intermediate polynomials?
```

Comment:
If it doesn't follow function calls, I think it's fair that it also does not follow references to intermediate columns.

Reply:
The only place this is used is in airgen. There is a comment "// TODO: links with next references can't be merged due to a witgen limitation." - maybe this is not the case any more?